pkg/operator/events/recorder.go

```diff
@@ -44,6 +44,8 @@ var podNameEnvFunc = func() string {
 
 // GetControllerReferenceForCurrentPod provides an object reference to a controller managing the pod/container where this process runs.
 // The pod name must be provided via the POD_NAME name.
+// Even if this method returns an error, it always return valid reference to the namespace. It allows the callers to control the logging
+// and decide to fail or accept the namespace.
 func GetControllerReferenceForCurrentPod(client kubernetes.Interface, targetNamespace string, reference *corev1.ObjectReference) (*corev1.ObjectReference, error) {
```

Comment:
need to update callsites to log error and continue instead of panicing.

Reply:
updated builder and installer pod, my goland don't see any more call sites in this repo.